test/gurka/test_traffic_routes.cc

```diff
@@ -0,0 +1,342 @@
+        } else {
+          historical[i] = 19 + (min_timestamp - 12);
+        }
+      }
+      return historical;
+    });
+  }
+
+  std::string make_request(const std::string& from,
+                           const std::string& to,
+                           std::string speed_types,
+                           bool prioritize_bidirectional,
+                           int date_time_type,
+                           std::string date_time_value) const {
+    const std::string query_pattern_with_speeds = R"({
+      "locations":[{"lat":%s,"lon":%s},{"lat":%s,"lon":%s}],
+      "costing": "auto",
+      "costing_options":{"auto":{"speed_types":[%s]}},
+      "prioritize_bidirectional":%s,
+      "date_time": { "type": "%d", "value": "%s" }
+    })";
+    return (boost::format(query_pattern_with_speeds) % std::to_string(map.nodes.at(from).lat()) %
+            std::to_string(map.nodes.at(from).lng()) % std::to_string(map.nodes.at(to).lat()) %
+            std::to_string(map.nodes.at(to).lng()) % speed_types %
+            std::to_string(prioritize_bidirectional) % date_time_type % date_time_value)
+        .str();
+  }
+
+  static gurka::map map;
+};
+
+gurka::map TrafficSmoothing::map = {};
+
+TEST_F(TrafficSmoothing, OneEdgeCurrentTime) {
+  std::string speeds = "\"freeflow\",\"constrained\",\"predicted\",\"current\"";
+
+  std::vector<valhalla::Api> results;
+  // invariant
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, false, 3, "current")));
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, true, 3, "current")));
+  // depart_at = now
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, false, 0, "current")));
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, true, 0, "current")));
+  // additional check, date_time_value should be ignored when type=0 (current)
+  results.emplace_back(
+      gurka::do_action(valhalla::Options::route, map,
+                       make_request("A", "B", speeds, false, 0, "2021-11-08T00:00")));
+  // depart_at = current_date_time
+  auto date_time = date_time_in_N_minutes(0);
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, false, 1, date_time)));
+  results.emplace_back(gurka::do_action(valhalla::Options::route, map,
+                                        make_request("A", "B", speeds, true, 1, date_time)));
+  for (const auto& result : results) {
+    // time_dependent_forward_a* is used for one-edge routes.
+    EXPECT_EQ(result.trip().routes(0).legs(0).algorithms(0), "time_dependent_forward_a*");
```

Comment:
why do we need this check ?

Reply:
sounds reasonable, no matter what the algo is, live traffic should be used